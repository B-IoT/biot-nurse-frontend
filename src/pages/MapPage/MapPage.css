.map-page {
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
}

.back-button {
  position: absolute;
  left: calc(5% - 20px);
}

.map-title {
  margin-top: 10px;
  z-index: 1000;
}

.top-container {
  align-items: center;
  justify-content: center;
  display: flex;
  position: relative;
  height: 125px;
  width: 100%;
  z-index: 999;
  background: var(--background);
}

.top-container-blur {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 20px;
  width: 120%;
  position: absolute;
  bottom: -10px;
  left: -10%;
  filter: blur(3px);
}

.top-container-shadow {
  height: 15px;
  width: 120%;
  position: absolute;
  background: var(--darker-shadow);
  filter: blur(1px);
}

.top-container-ambient {
  height: 50%;
  width: 100%;
  top: 0;
  position: absolute;
  background: var(--background);
}

.item-map-container {
  position: absolute;
  width: 100%;
  top: 125px;
  height: calc(100vh - 125px);
  overflow: hidden;
}
