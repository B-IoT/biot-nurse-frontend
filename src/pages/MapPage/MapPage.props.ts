export interface MapPageProps {
  /**
   * The item category
   */
  location: { state: { itemName: string } };
}

