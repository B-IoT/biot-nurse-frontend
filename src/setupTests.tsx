// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { BrowserRouter } from 'react-router-dom';
import React, { ComponentType } from 'react';

const queryClient = new QueryClient();
export const router = (children: any) => {
  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>{children}</BrowserRouter>
    </QueryClientProvider>
  );
};

const Wrapper: React.ComponentType = ({ children }) => {
  return router(children);
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper as ComponentType, ...options });

global.URL.createObjectURL = jest.fn();

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
