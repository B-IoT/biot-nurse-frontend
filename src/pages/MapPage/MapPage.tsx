import React from 'react';
import './MapPage.css';
import 'mapbox-gl/dist/mapbox-gl.css';

import BackButton from '../../components/button/BackButton/BackButton';
import ItemMap from '../../components/map/ItemMap/ItemMap';
import { MapPageProps } from './MapPage.props';

/**
 * The page displaying the location of the items corresponding to the given category.
 */
export default function MapPage(props: MapPageProps) {
  const { location } = props;

  return (
    <div className="map-page">
      <div className="top-container">
        <BackButton />
        <h1 className="map-title axiforma-semi-bold-blue-50px">
          {'Voici les ' + location.state.itemName + 's à proximité'}
        </h1>
      </div>
      <ItemMap itemName={location.state.itemName} />
    </div>
  );
}
