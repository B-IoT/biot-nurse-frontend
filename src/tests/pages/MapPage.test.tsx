import { render, router } from '../../setupTests';
import MapPage from '../../pages/MapPage/MapPage';
import renderer from 'react-test-renderer';
import React from 'react';

const location = { state: { itemName: 'Lit' } };

it('Map page renders without crashing', () => {
  render(<MapPage location={location} />);
});

it('Map page matches snapshot', () => {
  const rendered = renderer.create(router(<MapPage location={location} />));
  expect(rendered).toMatchSnapshot();
});
