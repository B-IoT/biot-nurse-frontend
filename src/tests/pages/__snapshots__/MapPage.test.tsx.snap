// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map page matches snapshot 1`] = `
<div
  className="map-page"
>
  <div
    className="top-container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="back-button css-1i20ack-Hover e128ygwg0"
      >
        <div
          className="back-pressed css-1gvt6rq-FadeIn e128ygwg3"
        />
        <div
          className="back-shadow css-1ypgtc4-FadeOut e128ygwg2"
        />
        <div
          className="back-light css-1ypgtc4-FadeOut e128ygwg2"
        />
        <div
          className="back-text axiforma-book-normal-blue-30px css-zcw12y-Scale e128ygwg1"
        >
          &lt; Retour
        </div>
      </div>
    </a>
    <h1
      className="map-title axiforma-semi-bold-blue-50px"
    >
      Voici les Lits à proximité
    </h1>
  </div>
  <div
    className="map-total-container"
  >
    <div
      className="map-control-left"
    >
      <div
        className="round-container css-1i20ack-Hover e128ygwg0"
      >
        <div
          className="round-pressed css-1gvt6rq-FadeIn e128ygwg3"
        />
        <button
          className="round-button-container"
          data-testid="round-button"
          onClick={[Function]}
        >
          <div
            className="round-button-container css-1ypgtc4-FadeOut e128ygwg2"
          >
            <div
              className="round-light"
            />
            <div
              className="round-shadow"
            />
          </div>
        </button>
        <div
          className="icon-container clear css-zcw12y-Scale e128ygwg1"
        >
          <img
            alt="Button icon"
            className="round-icon"
            src="navbarIcons/floorUp.svg"
          />
        </div>
      </div>
      <div
        className="round-input"
      >
        <img
          alt="Round input"
          className="round-input-img"
          src="roundInput.png"
        />
        <div>
          <input
            className="round-input-text axiforma-book-normal-blue-30px"
            onChange={[Function]}
            onKeyPress={[Function]}
            value="0"
          />
        </div>
      </div>
      <div
        className="round-container css-1i20ack-Hover e128ygwg0"
      >
        <div
          className="round-pressed css-1gvt6rq-FadeIn e128ygwg3"
        />
        <button
          className="round-button-container"
          data-testid="round-button"
          onClick={[Function]}
        >
          <div
            className="round-button-container css-1ypgtc4-FadeOut e128ygwg2"
          >
            <div
              className="round-light"
            />
            <div
              className="round-shadow"
            />
          </div>
        </button>
        <div
          className="icon-container clear css-zcw12y-Scale e128ygwg1"
        >
          <img
            alt="Button icon"
            className="round-icon"
            src="navbarIcons/floorDown.svg"
          />
        </div>
      </div>
    </div>
    <div
      className="map-container"
    >
      <div
        className="map-mask"
      >
        <div
          className="hidden"
          data-testid="hidden"
        >
          <div
            style={
              Object {
                "cursor": "grab",
                "height": "100vh",
                "position": "relative",
                "width": "90vw",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <div
                className=""
                style={
                  Object {
                    "height": "100%",
                    "overflow": "hidden",
                    "position": "absolute",
                    "visibility": "inherit",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="mask-edges clear"
        />
        <div
          className="blurred-edges clear"
        />
      </div>
      <div
        className="map-blur clear"
      >
        <div
          className="map-mask clear"
        >
          <div
            className="map-ambient clear"
          />
          <div
            className="map-shadow clear"
          />
          <div
            className="map-light clear"
          />
        </div>
      </div>
    </div>
    <div
      className="map-control-right"
    >
      <div
        className="round-container css-1i20ack-Hover e128ygwg0"
      >
        <div
          className="round-pressed css-1gvt6rq-FadeIn e128ygwg3"
        />
        <button
          className="round-button-container"
          data-testid="round-button"
          onClick={[Function]}
        >
          <div
            className="round-button-container css-1ypgtc4-FadeOut e128ygwg2"
          >
            <div
              className="round-light"
            />
            <div
              className="round-shadow"
            />
          </div>
        </button>
        <div
          className="icon-container clear css-zcw12y-Scale e128ygwg1"
        >
          <img
            alt="Button icon"
            className="round-icon"
            src="navbarIcons/zoom.svg"
          />
        </div>
      </div>
      <div
        className="round-container css-1i20ack-Hover e128ygwg0"
      >
        <div
          className="round-pressed css-1gvt6rq-FadeIn e128ygwg3"
        />
        <button
          className="round-button-container"
          data-testid="round-button"
          onClick={[Function]}
        >
          <div
            className="round-button-container css-1ypgtc4-FadeOut e128ygwg2"
          >
            <div
              className="round-light"
            />
            <div
              className="round-shadow"
            />
          </div>
        </button>
        <div
          className="icon-container clear css-zcw12y-Scale e128ygwg1"
        >
          <img
            alt="Button icon"
            className="round-icon"
            src="navbarIcons/dezoom.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
